mlir/lib/Dialect/LLVMIR/IR/LLVMDialect.cpp

```diff
@@ -2555,6 +2555,26 @@ Region *LLVMFuncOp::getCallableRegion() {
   return &getBody();
 }
 
+//===----------------------------------------------------------------------===//
+// UndefOp.
+//===----------------------------------------------------------------------===//
+
+/// Fold a undef operation to a dedicated undef attribute.
+OpFoldResult LLVM::UndefOp::fold(FoldAdaptor) {
+  // TODO: Should fold into ub::*?
```

Comment:
We want to stay in LLVM dialect world here. So for now I would fold to our own LLVM attributes! Same for Poison.